config/samples/microservices/kustomization.yaml

```diff
@@ -6,6 +6,9 @@ secretGenerator:
 - name: kbs-auth-public-key
   files:
   - kbs.pem
+- name: kbs-client
+  files:
+  - privateKey
```

Comment:
just a curiosity, which is the use case for storing the client private key in the kbs?

Reply:
@lmilleri,  we need to store the privateKey in a K8s Secret, such that we can later access it and use it to approach KBS Rest API. 

Storing it in the `kbs-operator-system` ns was chosen to allow keeping all kbs-operator resources in one place.

Reply:
We can also add a disclaimer that it's assumed the K8s environment is trusted and hence to private key is also stored as K8s secret. In fact the auth key pair can also be generated by cert-manager and stored as secret - https://cert-manager.io/docs/usage/certificate/

Reply:
So if I'm not wrong, any client able to attest himself can get the private key and later it could add/remove additional secrets and measurements to the kbs. Is that safe? For fetching resources we wouldn't need the private key.

Reply:
@lmilleri, 
Please note, kbs-operator is running on some cluster. 
This cluster must be trusted to run kbs :)  (Since the keys are stored in this this kbs which is running on that cluster.)
 
So if anyone is using kbs-operator to deploy kbs on some cluster, it seems that he trusts that cluster. Therefore we should have no concern to store a private key allowing access to the KBS secrets in a K8s Secret on that same cluster in the same namespace. 

> So if I'm not wrong, any client able to attest himself can get the private key and later it could add/remove additional secrets and measurements to the kbs. Is that safe? For fetching resources we wouldn't need the private key.

What do you means "any client"?
If you mean any pod, or any peer-pod, the answer is no. 
A pod (or a peer-pod)only have access to whatever the cluster admin give him access to and should not have access to a kbs-operator namespace at all and certainly not the keys on that namespace. 

Also, one may use two different clusters here - a first cluster to run the kbs-operator (a cluster which he trusts) and another cluster to run the peer pods (a cluster which he does not trust).

Reply:
> We can also add a disclaimer that it's assumed the K8s environment is trusted and hence the private key is also stored as K8s secret. 

This PR does not change anything that requires such a disclaimer in my view. 
Still, we should have security recommendations clarifying that KBS should be running on a trusted environment.
This also applies to the kbs-operator naturally.